### DISCUSSION TITLE
Inherit tags from abstract base classes: Is it possible?

### DISCUSSION THREAD

**Discussion by cristalp** [smallrye/smallrye-open-api#1897] [Q&A] [answered]
We have a lot of REST resources that use an abstract base class. And this base class has a tag which should be valid for all corresponding REST resources.

Coming from Swagger annotations and another Maven plugin, I'm finding that this doesn't work anymore with Smallrye. I have to annotate `@Tag` on every REST resource that inherits from the abstract base class (that already has the tag).

Am I doing something wrong or is this the way that tags work with the Smallrye Maven plugin?

**Comment by MikeEdgar** [comment]
You're right that `@Tag` should be inherited by sub-classes. If it's not working that way, sounds like a bug.

**Comment by cristalp** [comment]
Ok, I'll try to reproduce it with a very simple case and then write a bug.

Btw: Thanks a lot, you're very helpful to newbies like me  :-)

**Comment by cristalp** [comment, marked as answer]
Bug: https://github.com/smallrye/smallrye-open-api/issues/1901
